website/src/pages/api/messages/[id]/index.ts

```diff
@@ -2,12 +2,48 @@ import { withoutRole } from "src/lib/auth";
 import { createApiClientFromUser } from "src/lib/oasst_client_factory";
 import { getBackendUserCore } from "src/lib/users";
 
-const handler = withoutRole("banned", async (req, res, token) => {
+const getMessage = async (req, res, token) => {
+  const { id } = req.query;
+  try {
+    const user = await getBackendUserCore(token.sub);
+    const client = createApiClientFromUser(user);
+    const message = await client.fetch_message(id, user);
+    res.status(200).json(message);
+  } catch (e) {
+    res.status(500).json({ message: "Failed to fetch message" });
+  }
+};
+
+const deleteMessage = async (req, res, token) => {
   const { id } = req.query;
-  const user = await getBackendUserCore(token.sub);
-  const client = createApiClientFromUser(user);
-  const message = await client.fetch_message(id as string, user);
-  res.status(200).json(message);
+
+  if (token.role !== "admin") {
+    res.status(403).end();
+    return;
+  }
+
+  try {
+    const user = await getBackendUserCore(token.sub);
+    const client = createApiClientFromUser(user);
+    await client.delete_message(id);
+    res.status(200).json({ message: "Message deleted" });
+  } catch (e) {
+    res.status(500).json({ message: "Failed to delete message" });
+  }
+};
+
+const handlers = {
+  GET: getMessage,
```

Comment:
I feel like this is kinda confusing.  I would much rather we have a pure delete API route in `website/src/pages/api/admin/delete_message.ts`.

That way you don't need all this meta handler wrappers and the handler can directly use `withRole('admin')`.